verification/curator-service/api/test/model/parser.test.ts

```diff
@@ -0,0 +1,31 @@
+import { ParserDocument, parserSchema } from '../../src/model/parser';
+
+import { Error } from 'mongoose';
+import fullModel from './data/parser.full.json';
+import mongoose from 'mongoose';
+
+const Parser = mongoose.model<ParserDocument>('Parser', parserSchema);
+
+describe('validate', () => {
+    it('a parser without an AWS lambda ARN is invalid', async () => {
+        const missingArn = { ...fullModel };
+        delete missingArn.awsLambdaArn;
+
+        return new Parser(missingArn).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
+    });
+
+    it('a parser with a misformated AWS lambda ARN is invalid', async () => {
+        const badArn = { ...fullModel };
```

Comment:
up to you but can optionally inline as:

new Parser({ ...fullModel, awsLambdaArn: 'invalid...' });

not at all necessary if you think this is clearer, i just love to inline

Reply:
Inline'd!